File: Backend/src/main/java/com/example/backend/model/Post.java
package com.example.backend.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.sun.istack.NotNull;
import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
@AllArgsConstructor
@RequiredArgsConstructor
@NoArgsConstructor
@Table(name = "post")
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    private String title;

    @NotNull
    @Column(columnDefinition="TEXT")
    private String content;

    @NotNull
    private String author;

    @NotNull
    @JsonFormat(pattern="dd-MM-yyyy HH:mm")
    private LocalDateTime createdAt;

    @JsonFormat(pattern="dd-MM-yyyy HH:mm")
    private LocalDateTime updatedAt;

    private String tag;

    @OneToMany(mappedBy = "post", cascade = CascadeType.ALL)
    private Set<Comment> comments;

}

File: Frontend/src/app/model/Post.ts
export class Post {
    id!: number
    author!: string
    title!: string
    content!: string
    createdAt!: string
    updatedAt!: string
    thumbUpCount!: number
    thumbDownCount!: number
    rate!: number
    comments!: Comment[]
    tag!: string

    // constructor(
    //     id: number,
    //     author: string,
    //     title: string,
    //     content: string,
    //     createdAt: string,
    //     updatedAt: string,
    //     thumbUpCount: number,
    //     thumbDownCount: number,
    //     rate: number
    // ) {
    //     this.id = id;
    //     this.author = author;
    //     this.title = title;
    //     this.content = content;
    //     this.createdAt = createdAt;
    //     this.updatedAt = updatedAt;
    //     this.thumbUpCount = thumbUpCount;
    //     this.thumbDownCount = thumbDownCount;
    //     this.rate = rate;
    // }
}